xrpl4j-crypto-parent/xrpl4j-crypto-bouncycastle/src/main/java/org/xrpl/xrpl4j/crypto/signing/SingleKeySignatureService.java

```diff
@@ -0,0 +1,182 @@
+import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters;
+import org.bouncycastle.crypto.params.Ed25519PublicKeyParameters;
+import org.bouncycastle.crypto.signers.ECDSASigner;
+import org.bouncycastle.crypto.signers.Ed25519Signer;
+import org.bouncycastle.crypto.signers.HMacDSAKCalculator;
+import org.xrpl.xrpl4j.codec.addresses.UnsignedByteArray;
+import org.xrpl.xrpl4j.codec.binary.XrplBinaryCodec;
+import org.xrpl.xrpl4j.crypto.BcKeyUtils;
+import org.xrpl.xrpl4j.crypto.KeyMetadata;
+import org.xrpl.xrpl4j.crypto.KeyStoreType;
+import org.xrpl.xrpl4j.crypto.PrivateKey;
+import org.xrpl.xrpl4j.crypto.PublicKey;
+import org.xrpl.xrpl4j.keypairs.EcDsaSignature;
+import org.xrpl.xrpl4j.keypairs.HashUtils;
+import org.xrpl.xrpl4j.keypairs.Secp256k1;
+import org.xrpl.xrpl4j.model.jackson.ObjectMapperFactory;
+
+import java.math.BigInteger;
+import java.security.KeyStore;
+import java.util.Objects;
+
+/**
+ * <p>A {@link SignatureService} that holds a single private key, in-memory, using BouncyCastle as the underlying
+ * crypto implementation.</p>
+ *
+ * <p>WARNING: This implementation _might_ be appropriate for Android use, but should likely not be used in a
+ * server-side context. In general, prefer an implementation that offers a higher level of security.</p>
+ */
+public class SingleKeySignatureService extends AbstractSignatureService implements SignatureService {
+
+  private static final KeyStoreType KEY_STORE_TYPE = KeyStoreType.fromKeystoreTypeId("in-memory-single-key");
+
+  private final Ed25519Signer ed25519Signer;
+  private final ECDSASigner ecdsaSigner;
+  private final PrivateKey privateKey;
+
+  /**
+   * Required-args Constructor for use in development mode.
+   *
+   * @param privateKey A {@link KeyStore} to load all private keys from.
+   */
+  public SingleKeySignatureService(final PrivateKey privateKey) {
+    this(
+      new SignatureUtils(ObjectMapperFactory.create(), new XrplBinaryCodec()),
+      new Ed25519Signer(),
+      new ECDSASigner(new HMacDSAKCalculator(new SHA256Digest())),
+      privateKey
+    );
+  }
+
+  /**
+   * Required-args Constructor.
+   *
+   * @param signatureUtils An {@link SignatureUtils}.
+   * @param ed25519Signer  An {@link Ed25519Signer}.
+   * @param ecdsaSigner    An {@link ECDSASigner}.
+   * @param privateKey     A {@link PrivateKey} for obtain keys from.
+   */
+  public SingleKeySignatureService(
+    final SignatureUtils signatureUtils,
+    final Ed25519Signer ed25519Signer,
+    final ECDSASigner ecdsaSigner,
+    final PrivateKey privateKey
+  ) {
+    super(KEY_STORE_TYPE, signatureUtils);
+    this.ed25519Signer = Objects.requireNonNull(ed25519Signer);
+    this.ecdsaSigner = Objects.requireNonNull(ecdsaSigner);
+    this.privateKey = Objects.requireNonNull(privateKey);
+  }
+
+  @Override
+  public PublicKey getPublicKey(final KeyMetadata privateKeyMetadata) {
+    Objects.requireNonNull(privateKeyMetadata);
```

Comment:
Get rid of this so we can call with null

Reply:
@nkramer44 I experimented a bit with this change as well. No surprise to you, but I consider `null` to be a bug, so I propose this change instead (https://github.com/XRPLF/xrpl4j/commit/0fec0f30ffe871374a027190dc7af98ddacf0c16). 

However, we could alternatively introduce default method in `TransactionSigner` and `TransactionVerifier` that only takes a `Transaction` and then delegates to the method that takes both metadata+transaction. Only reason I didn't go that route for now is that it seemed to complicate the interface for a use-case that maybe we want to prod developers not to use (i.e., a single-key, in-memory implementation). But open to other ideas here if you don't like my proposal above.

Reply:
I think what you have is good for now.  We could always make a non-breaking change down the line to add that to `TransactionSigner` and `TransactionVerifier`, but it doesn't seem worth it now.